### ISSUE TITLE
Team-level prefix scan not supported in hierarchical parallelism

### ISSUE THREAD

**Issue by PointKernel** [kokkos/kokkos#3197]
I'm testing the team-level `parallel_scan` as shown below:

```C++
#include <Kokkos_Core.hpp>
​
#include <iostream>
​
constexpr int32_t N = 512;
​
using Kokkos::TeamPolicy;
using member_type = TeamPolicy<Kokkos::Cuda>::member_type;
​
int main() {
  Kokkos::initialize();
  {
    Kokkos::View<int32_t *> a_d("a_d", N);
    auto a_h = Kokkos::create_mirror_view(a_d);
​
    TeamPolicy<> policy = TeamPolicy<Kokkos::Cuda>(N / 32, 32);
    Kokkos::parallel_for(
        policy, KOKKOS_LAMBDA(const member_type &team) {
          auto teamRank = team.team_rank();
          auto leagueRank = team.league_rank();
​
          a_d(leagueRank * 32 + teamRank) = leagueRank * 32 + teamRank;
          team.team_barrier();
​
          Kokkos::parallel_scan(Kokkos::TeamThreadRange(team, leagueRank * 32,
                                                        leagueRank * 32 + 32),
                                [&](int i, int32_t &val, const bool final) {
                                  val += a_d(i);
                                  if (final)
                                    a_d(i) = val;
                                });
        });
​
    Kokkos::deep_copy(a_h, a_d);
​
    for (int i = 0; i < N / 32; i++) {
      for (int j = 0; j < 32; j++) {
        std::cout << a_h[i * 32 + j] << " ";
      }
      std::cout << "\n";
    }
  }
  Kokkos::finalize();
  return 0;
}
```

It's similar to https://github.com/kokkos/kokkos/blob/master/core/unit_test/TestTaskScheduler.hpp#L369, but I got the compilation error:
```
/global/homes/y/yswang/pps-kernel/kokkos/bin/nvcc_wrapper -I./ -I/global/homes/y/yswang/pps-kernel/kokkos/core/src -I/global/homes/y/yswang/pps-kernel/kokkos/containers/src -I/global/homes/y/yswang/pps-kernel/kokkos/algorithms/src -I/global/homes/y/yswang/pps-kernel/kokkos/core/src/eti --std=c++11 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -expt-extended-lambda -arch=sm_70 -I./ -I/global/homes/y/yswang/pps-kernel/kokkos/core/src -I/global/homes/y/yswang/pps-kernel/kokkos/containers/src -I/global/homes/y/yswang/pps-kernel/kokkos/algorithms/src -I/global/homes/y/yswang/pps-kernel/kokkos/core/src/eti -O3 -Werror=missing-braces -DREFDATA_TWOJ=2 -Dcuda_space -lineinfo  -c test.cpp
test.cpp(25): error: no instance of overloaded function "Kokkos::parallel_scan" matches the argument list
            argument types are: (Kokkos::Impl::TeamThreadRangeBoundariesStruct<signed int, Kokkos::Impl::CudaTeamMember>, lambda [](int, int32_t &, __nv_bool)->void)
```

**Comment by ndellingwood** [comment]
Cross-reference #45

**Comment by ndellingwood** [comment]
Should be addressed by #3536
